.gitlab/source_test/golang_deps_diff.yml

```
@@ -63,3 +63,16 @@ golang_deps_send_count_metrics:
     # Get API key to send metrics
     - DD_API_KEY=$($CI_PROJECT_DIR/tools/ci/fetch_secret.sh $AGENT_API_KEY_ORG2 token) || exit $?; export DD_API_KEY
     - dda inv -e go-deps.send-count-metrics --git-sha "${CI_COMMIT_SHA}" --git-ref "${CI_COMMIT_REF_NAME}"
+    
+golang_deps_test:
+  stage: source_test
+  image: registry.ddbuild.io/ci/datadog-agent-buildimages/deb_x64$DATADOG_AGENT_BUILDIMAGES_SUFFIX:$DATADOG_AGENT_BUILDIMAGES
+  tags: ["arch:amd64"]
+  rules:
+    - when: on_success
```

```suggestion
    - !reference  [.except_mergequeue]
    - when: on_success
```

In addition, should this be triggered only if go files change?